core/message.go

```diff
@@ -73,12 +73,12 @@ func (meta MetaData) SetValue(key string, value []byte) {
 }
 
 // GetValue returns a meta data value by key
-func (meta MetaData) GetValue(key string, defaultValue []byte) []byte {
+func (meta MetaData) GetValue(key string) []byte {
 	if value, isSet := meta[key]; isSet {
 		return value
 	}
 
-	return defaultValue
+	return []byte{}
```

Comment:
TBD
Maybe returning nil is a better idea here?
The thing is - you cannot distinguish between an empty value and a non-existing value here.
On the other hand if you always return a "valid" result you don't need to check for errors ...